lib/haml/helpers/action_view_mods.rb

```diff
@@ -55,7 +55,9 @@ module TagHelper
       def content_tag_with_haml(name, *args, &block)
         return content_tag_without_haml(name, *args, &block) unless is_haml?
 
-        preserve = haml_buffer.options.fetch(:preserve, %w[textarea pre code]).include?(name.to_s)
+        @_content_tag_name_cache ||= {}
+        name_string = @_content_tag_name_cache[name] ||= name.to_s
```

Comment:
Idea here being that we only needed to run `to_s` on each content tag once. In particular, I was getting `a` in here a lot from links. Any better ideas are welcome.

Reply:
Performance is within margin of error but saves a string allocation for each time this method is called.

```ruby
Warming up --------------------------------------
      master_version   188.121k i/100ms
        fast_version   201.722k i/100ms
Calculating -------------------------------------
      master_version      5.176M (±10.1%) i/s -     25.584M in   5.007528s
        fast_version      5.595M (± 7.3%) i/s -     27.838M in   5.004395s

Comparison:
        fast_version:  5595000.8 i/s
      master_version:  5176423.8 i/s - same-ish: difference falls within error
```

Reply:
Having `%w[textarea pre code]` as constant would be enough. I guess main performance gain comes from that. I'm reluctant to introduce ivar in template evaluation environment, and in general cache is likely to introduce bug.

Reply:
@k0kubun I went ahead and added the constant for `%w[textarea pre code] in https://github.com/haml/haml/pull/966. Additionally, what do you think about adding a constant for common `content_tag` options to cache the `name.to_s` call? Mainly the things that show up in [this]( https://github.com/rails/rails/search?l=Ruby&q=content_tag&type=&utf8=%E2%9C%93) search of the Rails repo and any other tag suggestions you have?

Reply:
And your above benchmark is now expired after you add constant, so you should put benchmark again. 

As I said,

> I'm reluctant to introduce ivar in template evaluation environment, and in general cache is likely to introduce bug.

So at least I wouldn't merge it unless there's SIGNIFICANT performance benefit in benchmark. Basically `to_s` is not slow.